File: app/src/main/java/rod/bailey/trafficatnsw/hazard/details/LineListItemView.java
package rod.bailey.trafficatnsw.hazard.details;

import android.content.Context;
import android.widget.TextView;

import static android.graphics.Color.*;
import static android.graphics.Typeface.*;
import static android.text.TextUtils.TruncateAt.*;
import static rod.bailey.trafficatnsw.util.DisplayUtils.*;

public class LineListItemView extends TextView {

	private static final int TEXT_SIZE_SP = 12;
	@SuppressWarnings("unused")
	private static final String TAG = LineListItemView.class.getSimpleName();

	public LineListItemView(Context ctx, String text) {
		super(ctx);

		setText(text);
		setTextColor(BLACK);
		setTypeface(DEFAULT);
		// Note that setTextSize() takes an argument in SP, not PX
		setTextSize(TEXT_SIZE_SP);
		setSingleLine();
		setEllipsize(END);
		setBackgroundColor(TRANSPARENT);
		setBackgroundColor(WHITE);

		int p = dp2Px(ctx, 5);
		setPadding(p, p, p, p);
	}
}

File: app/src/main/java/rod/bailey/trafficatnsw/hazard/details/view/LineListItemView.kt
package rod.bailey.trafficatnsw.hazard.details.view

import android.content.Context
import android.support.v7.widget.AppCompatTextView
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EViewGroup
import org.androidannotations.annotations.ViewById
import rod.bailey.trafficatnsw.R

@EViewGroup(R.layout.list_item_hazard_detail_line)
open class LineListItemView(ctx: Context,  val lineOfText: String) : FrameLayout(ctx) {

	@ViewById(R.id.tv_hazard_detail_line)
	@JvmField
	var textView: AppCompatTextView? = null

	@AfterViews
	fun afterViews() {
		textView?.text = lineOfText
	}
}
